webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/controllers/VolumeTracingController.scala

```diff
@@ -203,6 +219,159 @@ class VolumeTracingController @Inject()(val tracingService: VolumeTracingService
+                (tracing.boundingBox.width + tracing.boundingBox.topLeft.x) / magParsed.x,
+                (tracing.boundingBox.height + tracing.boundingBox.topLeft.y) / magParsed.y,
+                (tracing.boundingBox.depth + tracing.boundingBox.topLeft.z) / magParsed.z
+              ),
+              "dimension_seperator" -> "."
+            ))
+      }
+  }
+
+  def zGroup(token: Option[String], tracingId: String): Action[AnyContent] = Action.async { implicit request =>
+    accessTokenService.validateAccess(UserAccessRequest.readTracing(tracingId), urlOrHeaderToken(token, request)) {
+      Future(Ok(Json.obj("zarr_format" -> 2)))
+    }
+  }
+
+  def rawZarrCube(token: Option[String], tracingId: String, mag: String, cxyz: String): Action[AnyContent] =
+    Action.async { implicit request =>
+      {
+        val combinedToken = urlOrHeaderToken(token, request)
+        accessTokenService.validateAccess(UserAccessRequest.readTracing(tracingId), combinedToken) {
+          for {
+            tracing <- tracingService.find(tracingId) ?~> Messages("tracing.notFound")
+
+            existingMags = tracing.resolutions.map(v => Vec3Int(v.x, v.y, v.z))
+            magParsed <- Vec3Int.fromMagLiteral(mag, allowScalar = true) ?~> Messages("dataLayer.invalidMag", mag)
+            _ <- bool2Fox(existingMags.contains(magParsed)) ?~> Messages("tracing.wrongMag", tracingId, mag) ~> 404
+
+            (c, x, y, z) <- ZarrCoordinatesParser.parseDotCoordinates(cxyz) ?~> Messages("zarr.invalidChunkCoordinates") ~> 404
+            _ <- bool2Fox(c == 0) ~> Messages("zarr.invalidFirstChunkCoord") ~> 404
+            cubeSize = DataLayer.bucketLength
+            request = WebKnossosDataRequest(
+              position = Vec3Int(x * cubeSize * magParsed.x, y * cubeSize * magParsed.y, z * cubeSize * magParsed.z),
+              mag = magParsed,
+              cubeSize = cubeSize,
+              fourBit = Some(false),
+              applyAgglomerate = None,
+              version = None
+            )
+            (data, _) <- tracingService.data(tracingId, tracing, List(request))
+
+            dataWithFallback <- getFallbackLayerDataIfEmpty(tracing, data, mag, cxyz, combinedToken) ?~> "Getting fallback layer failed" ~> 404
+          } yield Ok(dataWithFallback).withHeaders()
+        }
+      }
+    }
+
+  private def getFallbackLayerDataIfEmpty(tracing: VolumeTracing,
+                                          data: Array[Byte],
+                                          mag: String,
+                                          cxyz: String,
+                                          urlToken: Option[String]): Fox[Array[Byte]] = {
+    val isAnnotationEmpty = data.forall(x => x == 0)
+
+    // Since we only do all bucket requests, complete bucket needs to empty so that we use fallback
+    if (isAnnotationEmpty) {
+      val organizationName = tracing.getOrganizationName
```

Comment:
getOrganizationName will return an empty string in case of a legacy annotation. Will your code work with emptystring? As I read it, getDataStoreURIForDataSource can actually handle the Option directly. However, `fallbackLayerBucket` probably can handle neither None nor emptystring (?) If that is true, I’d prefer a very early check asserting that the option is nonEmpty with a http 400 failure. If this is the case, zarr streaming is only availble for volume tracings that do include the organization name tag. (All created after november 2020 do)

Reply:
Since it's just calling the URL where an organisationName is expected, I think it's easier to just fail for now.